include/sharg/detail/format_parse.hpp

```diff
@@ -319,20 +319,22 @@ class format_parse : public format_base
 
         if (auto it = map.find(in); it == map.end())
         {
-            std::vector<std::pair<std::string_view, option_t>> key_value_pairs(map.begin(), map.end());
-            std::ranges::sort(key_value_pairs, [] (auto pair1, auto pair2)
-                {
-                    if constexpr (std::totally_ordered<option_t>)
-                    {
-                        if (pair1.second != pair2.second)
-                            return pair1.second < pair2.second;
-                    }
-                    return pair1.first < pair2.first;
-                });
-
-            throw user_input_error{seqan3::detail::to_string("You have chosen an invalid input value: ", in,
-                                                             ". Please use one of: ",
-                                                             key_value_pairs | std::views::keys)};
+            std::string keys = [&map] ()
+            {
+                std::vector<std::string> keys;
+                for (auto & [key, value] : map)
+                    keys.emplace_back(key);
+
+                std::sort(keys.begin(), keys.end()); // needed for deterministic output when using unordered maps
```

Comment:
We did it with pairs and a comparator, to ensure the values have a meaningful order.